src/Components/Components/src/ComponentRouteData.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Describes information determined during routing that specifies
+    /// the page to be displayed.
+    /// </summary>
+    public sealed class ComponentRouteData
```

Comment:
This could easily have been a `readonly struct`. I don't really have much reason for picking either way. Since these types are only instantiated once per user navigation event, the perf impact would be negligible.

Reply:
Classes are always a safer choice if we don't have a good reason to make a struct.